File: benchmarks/src/nodejs/higher_order.js
const base = [];
for (let i = 0; i < 1000; i++) base.push(i);
const R = 2000;
let acc = 0;
const start = Date.now();
for (let r = 0; r < R; r++) {
    const d = base.map(x => x * 2 + 1);
    const f = d.filter(x => x % 3 !== 0);
    const s = f.reduce((a, x) => a + x, 0);
    acc = (acc + s + f.length) % 1000003;
}
const end = Date.now();
console.log("Result: " + acc);
console.log("Time: " + Math.floor(end - start) + "ms");

File: benchmarks/src/perl/higher_order.pl
use Time::HiRes qw(time);
use List::Util qw(reduce);

my @base = (0..999);
my $R = 2000;
my $acc = 0;
my $start = time();
for (my $r = 0; $r < $R; $r++) {
    my @d = map { $_ * 2 + 1 } @base;
    my @f = grep { $_ % 3 != 0 } @d;
    my $s = reduce { $a + $b } 0, @f;
    $acc = ($acc + $s + scalar(@f)) % 1000003;
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "Result: $acc\n";
printf "Time: %.0fms\n", $ms;
